src/main/java/pl/edu/mimuw/Main.java

```diff
@@ -1,18 +1,34 @@
 package pl.edu.mimuw;
 
 import java.io.IOException;
+import java.sql.Timestamp;
+import java.util.Iterator;
+import java.util.stream.Stream;
 
 public class Main {
+  private static Stream<BankAction> actionsGenerator() {
+    Stream.Builder<pl.edu.mimuw.BankAction> builder = Stream.builder();
 
-  public static void main(String[] args) throws IOException {
+    // Adding elements in the stream of Strings
+    return builder.add(new DepositBankAction(100, Timestamp.valueOf("2022-03-01 12:00:00"), Timestamp.valueOf("2022-03-30 12:00:00"), 10))
+      .add(new DepositBankAction(200, Timestamp.valueOf("2022-04-01 12:00:00"), Timestamp.valueOf("2022-04-30 12:00:00"), 20))
+      .add(new DepositBankAction(300, Timestamp.valueOf("2022-04-01 12:00:00"), Timestamp.valueOf("2022-04-30 12:00:00"), 30))
+      .build();
+  }
+  public static void main(String[] args) {
     System.out.println("Welcome in MIM UW Bank");
     final var bank = new Bank();
 
-    // TODO: play with bank:
-    // - add new clients
-    // - add different actions on clients accounts
-    //    * try to do it with some bigger collections of actions (maybe even use streams)
+    BankClient jk = new BankClient("Jakub", "Klimek", 20, bank.newAccountNumber());
+    bank.addClient(jk);
+    BankClient mp = new BankClient("Maciej", "Procyk", 23, bank.newAccountNumber());
+    bank.addClient(mp);
+
+    jk.addAction(new LoanBankAction(5000, Timestamp.valueOf("2021-10-15 12:00:00"), Timestamp.valueOf("2022-10-15 12:00:00"), 5));
 
+    for(var ac : actionsGenerator().toList()) {
```

Comment:
lepiej `for(var ac : actionsGenerator().iterator())`
a nawer `actionsGenerator().forEach(mp::addAction)` (`::` służy żeby powiedzieć, że chodzi nam o referencje do metody)

Reply:
okazuje się że to `toList` jest z Javy 16 więc to nie nasza bajka niestety, zdecydoeanie zatem `iterator()`

Reply:
a przy okazji popraw ustawienia IDE w ProjectStructure, bo zaciąga Ci z systemu domyślną Jave 16 🙈

Reply:
Chyba nawet 17, na jakiej pracować w takim razie?

Reply:
na 11 co najwyżej z tego co wiem dobrze